include/SFML/System/Angle.hpp

```diff
@@ -0,0 +1,458 @@
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (a number)
+///
+/// \return \a left divided by \a right
+///
+////////////////////////////////////////////////////////////
+constexpr Angle& operator /=(Angle& left, float right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary / operator to compute the ratio of two angle values
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return \a left divided by \a right
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr float operator /(Angle left, Angle right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary % operator to compute remainder of an angle value
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return \a left modulo \a right
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator %(Angle left, Angle right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary %= operator to compute/assign remainder of an angle value
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return \a left modulo \a right
+///
+////////////////////////////////////////////////////////////
+constexpr Angle& operator %=(Angle& left, Angle right);
+
+inline namespace literals
+{
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief User defined literal for angles in degrees
+///
+/// \param angle Angle in degrees
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _deg(long double angle);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief User defined literal for angles in degrees
+///
+/// \param angle Angle in degrees
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _deg(unsigned long long angle);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief User defined literal for angles in radians
+///
+/// \param angle Angle in radians
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _rad(long double angle);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief User defined literal for angles in radians
+///
+/// \param angle Angle in radians
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _rad(unsigned long long angle);
```

Comment:
Maybe we should for each mention a short example usage, like
```cpp
sf::Angle angle = 22.5_deg;
```

This would also alleviate potential confusion with the parameter types `long double` and `unsigned long long`.